flink-tests/src/test/java/org/apache/flink/test/recovery/BatchFineGrainedRecoveryITCase.java

```diff
@@ -467,4 +509,66 @@ public void mapPartition(Iterable<Long> values, Collector<Long> out) throws Exce
 			}
 		}
 	}
+
+	private static class MiniClusterClient extends RestClusterClient<StandaloneClusterId> {
```

Comment:
Why do we need to extend the `RestClusterClient`? Wouldn't it be enough to simply use the `RestClient` with the rest address which one gets from the `MiniCluster`? Then one would not have to change the access modifiers of the `RestClusterClient` and not have to jump through all these hoops to set up the client configuration.

Reply:
True, there is actually not so much code to reuse with the `RestClusterClient`.